errors/errors.go

```diff
@@ -0,0 +1,90 @@
+package errors
+
+import (
+	"errors"
+	"time"
+)
+
+// NoRetryError wraps an error, preventing retries in the SDK.  This permanently
+// fails a step and function.
+func NoRetryError(err error) error {
+	return noRetryError{Err: err}
+}
+
+// IsNoRetryError returns whether an error is a NoRetryError
+func IsNoRetryError(err error) bool {
+	return errors.Is(err, noRetryError{})
+}
+
+// noRetryError represents an error that will not be retried.
+type noRetryError struct {
+	Err error
+}
+
+// Error fulfils the builtin go error interface.  This returns the original root cause
+// for debugging via logging - all logging interfaces & tracing packages use Error() to
+// capture error information.
+func (e noRetryError) Error() string {
+	return e.Err.Error()
+}
+
+// Unwrap fulfils errors.Unwrap, allowing us to use the builtin error functionality to
+// manage the original error, or to chain public errors.
+func (e noRetryError) Unwrap() error {
+	return e.Err
+}
+
+// Unwrap fulfils errors.Unwrap, allowing us to use the builtin error functionality to
+// manage the original error, or to chain public errors.
+func (e noRetryError) Is(target error) bool {
+	switch target.(type) {
+	case *noRetryError, noRetryError:
+		return true
+	default:
+		return false
+	}
+}
+
+// RetryAtError allows you to specify the time at which the next retry should occur. This
+// wraps your error, leaving the original cause and error message available.
+func RetryAtError(err error, at time.Time) error {
+	return retryAtError{Err: err, At: at}
+}
+
+// GetRetryAtTime returns the time from a retryAtError, or nil.
+func GetRetryAtTime(err error) *time.Time {
```

Comment:
Why do these need to be exported?  Isn't this internal to the SDK for response output?

Reply:
Clarified — circular imports.  Resolving!